plugins/woocommerce-admin/client/marketplace/components/feedback-modal/feedback-modal.tsx

```diff
@@ -0,0 +1,152 @@
+	const SUPPRESS_IF_DISMISSED_X_TIMES = 2; // if the user dismisses the snackbar this many times, stop asking for feedback
+	const SUPPRESS_IF_AFTER_DATE = '2024-01-01'; // if this date is reached, stop asking for feedback
+
+	// We don't want the "How easy was it to find an extension?" dialog to appear forever:
+	const FEEDBACK_DIALOG_CAN_APPEAR = new Date( SUPPRESS_IF_AFTER_DATE ) > new Date();
+	if( ! FEEDBACK_DIALOG_CAN_APPEAR ) {
+		return <></>;
+	}
+
+	const [ isOpen, setOpen ] = useState( false );
+	const [ thoughts, setThoughts ] = useState( '' );
+	const [ easyToFind, setEasyToFind ] = useState( 0 );
+	const [ easyToFindValidiationFailed, setEasyToFindValidiationFailed ] = useState( false );
+	const [ meetsMyNeeds, setMeetsMyNeeds ] = useState( 0 );
+	const [ meetsMyNeedsValidiationFailed, setMeetsMyNeedsValidiationFailed ] = useState( false );
+	const openModal = () => setOpen( true );
+	const closeModal = () => { incrementDismissedTimes(); setOpen( false ) };
+	const { createNotice } = useDispatch( 'core/notices' );
+
+	// eslint-disable-next-line react-hooks/exhaustive-deps -- [] => we only want this effect to run once, on first render
+	useEffect( maybeShowSnackbar, [] );
+
+	// Save that we dismissed the dialog or snackbar TODAY so we don't show it again until tomorrow (if ever)
+	const dismissToday = () => localStorage.setItem( LOCALSTORAGE_KEY_LAST_REQUESTED_DATE, ( new Date() ).toDateString() );
+
+	// Returns the number of times that the request for feedback has been dismissed
+	const dismissedTimes = () => parseInt ( localStorage.getItem( LOCALSTORAGE_KEY_DISMISSAL_COUNT ) || '0' );
+
+	// Increment the number of times that the request for feedback has been dismissed
+	const incrementDismissedTimes = () => { dismissToday(); localStorage.setItem( LOCALSTORAGE_KEY_DISMISSAL_COUNT, `${dismissedTimes() + 1}` ); }
+
+	// Dismiss forever (by incrementing the number of dismissals to a high number), e.g. when feedback is provided
+	const dismissForever = () => { dismissToday(); localStorage.setItem( LOCALSTORAGE_KEY_DISMISSAL_COUNT, `${SUPPRESS_IF_DISMISSED_X_TIMES}` ); }
+
+	// Returns true if dismissed forever (either by dismissing at least SUPPRESS_IF_DISMISSED_X_TIMES times, or by submitting feedback)
+	const isDismissedForever = () => dismissedTimes() >= SUPPRESS_IF_DISMISSED_X_TIMES;
+
+	function maybeShowSnackbar(){
+		// don't show if the user has already given feedback or otherwise suppressed:
+		if( isDismissedForever() ) {
+			return;
+		}
+
+		// don't show if the user has already declined to provide feedback today:
+		const today = ( new Date() ).toDateString();
+		if( today === localStorage.getItem( LOCALSTORAGE_KEY_LAST_REQUESTED_DATE ) ) {
+			return;
+		}
+
+		createNotice(
+			'success',
+			__( 'How easy is it to find an extension?' ),
+			{
+				type: 'snackbar',
+				icon: <>💬</>,
```

Comment:
🗒️  **Wrong icon!** I've just noticed that this isn't the icon that's called for in the design! I'll get to this.